### ISSUE TITLE
[padding-line-between-statements] wrong blank line in overloaded exported functions

### ISSUE THREAD

**Issue by Ulrikop** [typescript-eslint/typescript-eslint#4030]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/padding-line-between-statements": [
      "error",
      { "blankLine": "always", "prev": "*", "next": "block-like" },
     ]
  }
}
```

```TS
export function bar(arg: string): string;
export function bar(arg: number): number;
export function bar(arg: string | number) {
  return arg;
}
```

**Expected Result**

No error.

**Actual Result**

Error `Expected blank line before this statement.`

The auto fix creates following code:

```TS
export function bar(arg: string): string;
export function bar(arg: number): number;

export function bar(arg: string | number) {
  return arg;
}
```

**Additional Info**

Overloaded functions without an `export` or an overloaded methods of a class works like expected:

```TS
class Foo {
  bar(arg: string): string;
  bar(arg: number): number;
  bar(arg: string | number) {
    return arg;
  }
}

function bar(arg: string): string;
function bar(arg: number): number;
function bar(arg: string | number) {
  return arg;
}
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.1.0` |
| `@typescript-eslint/parser`        | `5.1.0` |
| `TypeScript`                       | `4.1.6` |
| `ESLint`                           | `8.0.0` |
| `node`                             | `14.18.1` |

**Comment by lonyele** [comment]
@Ulrikop Can you check the linked PR? It's the one that fixes the problem but in a way that you are not expecting. I also wrote some options that can be used to your need
